### DISCUSSION TITLE
How can I create an "infinite" canvas under zoom and pan?

### DISCUSSION THREAD

**Discussion by DrDryg** [AvaloniaUI/Avalonia#13100] [General]
I have a canvas with a number of objects that I can select and move around. I can also pan and zoom in my canvas. But here I encounter an issue. If I zoom out or pan I reach the border of my canvas. But I want my canvas to resize when I zoom out or pan. But I cannot figure out how to do it properly.

- I've tried to change the size of the canvas whenever I zoom, be listening to a zoomChanged event. (I use the pan and zoom control). But if I autosize the canvas based on the window size I cannot simply scale my canvas since its width and height are NaN. 
- Even if I set a fixed width and height which I like to avoid the behavior is not very good. The center of the zoom seems to jump around quite a bit. 
- Neither can I draw outside the Canvas with a good result. It seems to be no problem to draw there but there is a glitch when I draw a rectangle for object selection. Some of the rectangle is sometimes not cleared:

So It can look like this:
![image](https://github.com/AvaloniaUI/Avalonia/assets/36692294/1c222107-a735-4331-b452-90bb3faf79b9)

Anyone with some idea how to create the "infinite" canvas? Or why I get some leftovers if I draw outside my canvas in the ZoomBorder?

**Comment by Paegasus** [comment]
Did anyone figure it out?

**Comment by Shadowblitz16** [comment]
I would like to know this as well.
I am also wondering about arrows keys to pan though

**Comment by thevortexcloud** [comment]
FYI: There is an existing pan and zoom control available. 

https://github.com/wieslawsoltes/PanAndZoom
